dktest.go

```diff
@@ -167,42 +168,57 @@ func waitContainerReady(ctx context.Context, lgr logger, c ContainerInfo,
 
 // Run runs the given test function once the specified Docker image is running in a container
 func Run(t *testing.T, imgName string, opts Options, testFunc func(*testing.T, ContainerInfo)) {
+	err := RunContext(context.Background(), t, imgName, opts, func(containerInfo ContainerInfo) error {
+		testFunc(t, containerInfo)
+		return nil
+	})
+	if err != nil {
+		t.Fatal("Failed to", err)
+	}
+}
+
+// RunContext is similar to Run, but takes a parent context and returns an error and doesn't rely on a testing.T.
+func RunContext(ctx context.Context, logger Logger, imgName string, opts Options, testFunc func(ContainerInfo) error) (retErr error) {
```

Comment:
Though the main reason for the new method was to return an error, I thought it was a good opportunity to take a parent context as well, and the go stdlib has a few methods with the `Context` suffix, so avoids coming up with some name to indicate that it returns an error.